### PR TITLE
chore: introduce app context

### PR DESCRIPTION
**Pull request by mr25mr**
- app context only contains UI5 model and manifest details
- fix getting correct UI5 version on initial load
- place caching logic in a single file

### CODE REVIEW

File: packages/language-server/src/manifest-handling.ts [hunk 1]
```diff
@@ -135,6 +131,8 @@ async function readManifestFile(
   const flexEnabled = manifestJsonObject["sap.ui5"]?.flexEnabled;
   const minUI5Version =
     manifestJsonObject["sap.ui5"]?.dependencies?.minUI5Version;
-
-  return { flexEnabled: flexEnabled, minUI5Version: minUI5Version };
+  return {
+    flexEnabled: flexEnabled,
+    minUI5Version: minUI5Version,
+  };
```
Comment (UweSchwerk) [inline packages/language-server/src/manifest-handling.ts:137]: Maybe revert if it is only a change in formatting?

File: yarn.lock [hunk 2]
```diff
@@ -8841,11 +8841,6 @@ tr46@^1.0.1:
   dependencies:
     punycode "^2.1.0"
 
-tr46@~0.0.3:
-  version "0.0.3"
-  resolved "https://registry.yarnpkg.com/tr46/-/tr46-0.0.3.tgz#8184fd347dac9cdc185992f3a6622e14b9d9ab6a"
-  integrity sha512-N3WMsuqV66lT30CrXNbEjx4GEwlow3v6rr4mCcv6prnfwhS01rkgyFdjPNBYd9br7LpXV1+Emh01fHnq2Gdgrw==
-
```
Comment (UweSchwerk) [inline yarn.lock:8848]: any explanation why this is removed from yarn.lock ?

**Review by UweSchwerk — COMMENTED** [review]
overall this looks good to me (I was just checking code style, not functional correctness). 
Pls also sign Contributor License Agreement before removing draft flag.

File: packages/language-server/src/cache.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  constructor() {
+    this._manifestDetails = new Map();
+    this._ui5YamlDetails = new Map();
+    this._ui5Model = new Map();
+  }
+
+  /**
+   * Get all cached manifest details
+   */
+  getAllManifestDetails(): Map<AbsolutePathToManifest, ManifestDetails> {
```
Comment (vadson71) [inline packages/language-server/src/cache.ts:22]: I'm not sure we need all these getters and setters. 
They usually not needed unless we have some internal data transformation/validation inside the class.
You allow to access _manifestDetails directly and at the same time offer to update it via setManifestDetails method. IMHO it is fake security. Much simpler just to have public class member

File: packages/language-server/src/manifest-handling.ts (hunk 1 above)
Comment (mr25mr) [reply packages/language-server/src/manifest-handling.ts:137]: with current formatting and running ```format:validate```  shows every thing is compliant with Prettier code style.. So it should be fine,

File: packages/language-server/src/cache.ts (hunk 3 above)
Comment (mr25mr) [reply packages/language-server/src/cache.ts:22]: well we are not very much concern about security here, but anyway to to make private class members not to be publicly exposed, redefine it. Thanks anyway.

File: yarn.lock (hunk 2 above)
Comment (mr25mr) [reply yarn.lock:8848]: well. It is managed by yarn and I am not sure if that is needed to be in yarn,lock file.

**Comment by mr25mr** [comment]
covered in this PR https://github.com/SAP/ui5-language-assistant/pull/523

**Pull request closed without merging**